### PR TITLE
Fix feature flag not getting created with same team and key after deleting it

### LINKED ISSUE #2421: Feature Flag cannot recreate with the same key after deleting
## Bug description
Feature Flag cannot recreate with the same key after deleting 

Steps to reproduce: 

1. create feature flag with key `bla`
2. delete the feature flag
3. try to create again feature flag with `bla` key
4. alert will popup with This key already exists
image

## Expected behavior
New feature flag will be added

## Environment

self-hosting

### PR DESCRIPTION
**Pull request by 0xPiranhaCodes**
## Changes

Before creating a new feature flag, checking if there are any objects with the same team and key which is marked as deleted, if so then deleting them permanently

closes #2421 

## Checklist

- [  ] All querysets/queries filter by Organization, Team, and User (if this PR affects ANY querysets/queries).
- [  ] Django backend tests (if this PR affects the backend).
- [  ] Cypress end-to-end tests (if this PR affects the frontend).

### CODE REVIEW

**Comment by timgl** [comment]
Thanks for doing this! Looks like another contributor raised the same pr (#2462) at almost the same time, however you were first so I'd like to give you preference. Also your PR does correctly filter by team. We will need to have a test for this behaviour though (see the other PR for an example).

Thanks!

**Comment by 0xPiranhaCodes** [comment]
@timgl Thanks for giving me the preference.

**Comment by 0xPiranhaCodes** [comment]
@timgl I added a commit with test for this behaviour. Can you review it and let me know if there are any issues.

File: posthog/api/test/test_feature_flag.py [hunk 1]
```diff
@@ -150,3 +150,16 @@ def test_cannot_delete_feature_flag_on_another_team(self, mock_capture):
         self.assertTrue(FeatureFlag.objects.filter(pk=self.feature_flag.pk).exists())
 
         mock_capture.assert_not_called()
+
+    def test_creating_a_feature_flag_with_same_team_and_key_after_deleting(self):
+        instance = FeatureFlag.objects.create(team=self.team, created_by=self.user, key="alpha-feature")
+        instance.delete()
```
Comment (timgl) [inline posthog/api/test/test_feature_flag.py:156]: This doesn't test the right thing. You can just pass `deleted=True` to the line above.

File: posthog/api/feature_flag.py [hunk 2]
```diff
@@ -44,6 +44,7 @@ def create(self, validated_data: Dict, *args: Any, **kwargs: Any) -> FeatureFlag
         validated_data["created_by"] = request.user
         validated_data["team"] = request.user.team
         try:
+            FeatureFlag.objects.filter(key=validated_data["key"], team=request.user.team, deleted=True).delete()
```
Comment (timgl) [inline posthog/api/feature_flag.py:47]: The same thing should happen on `def update()` below.

**Review by timgl — COMMENTED** [review]
Almost there! A few small comments

**Comment by 0xPiranhaCodes** [comment]
> Almost there! A few small comments

@timgl thanks for the review, I have resolved the comments you have mentioned.

**Comment by timgl** [comment]
@satheesh1997 Thank you and congratulations on your first PR!

cc @yakkomajuri for merch

**Comment by weyert** [comment]
@satheesh1997 Thank you for fixing this issue!

**Comment by yakkomajuri** [comment]
Hey @satheesh1997! Would love to send some PostHog merch over to you for this PR. Could you email me at yakko[at]posthog[.]com so I have your contact?

**Comment by 0xPiranhaCodes** [comment]
> yakko[at]posthog[.]com

Sure, sending you an email

**Pull request merged**